examples/fishnet_api/main.py

```diff
@@ -188,11 +253,11 @@ async def upload_algorithm(algorithm: UploadAlgorithmRequest) -> Algorithm:
     """
     if algorithm.id_hash is not None:
         # update existing algorithm
-        resp = (await Algorithm.fetch(algorithm.id_hash))
+        resp = await Algorithm.fetch(algorithm.id_hash)
         old_algorithm = resp[0] if resp else None
         if old_algorithm is not None:
             if old_algorithm.owner != algorithm.owner:
-                raise Exception("Cannot overwrite algorithm that is not owned by you")
+                raise HTTPException(status_code=404, detail="Cannot overwrite algorithm that is not owned by you")
```

Comment:
Should be 403 "Forbidden", check out the [official HTTP status codes documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) for more info.
```suggestion
                raise HTTPException(status_code=403, detail="Cannot overwrite algorithm that is not owned by you")
```